helm-chart/amalthea-sessions/templates/_helpers.tpl

```diff
@@ -1,3 +1,11 @@
+{{/*
+Check consistency requirements.
+*/}}
+{{- $expectedName := printf "%s-%s" .Values.csiRclone.storageClassName "secret-annotation" -}}
+{{- if not eq .Values.rcloneStorageClass $expectedName -}}
+{{- fail "ERROR: .Values.rcloneStorageClass does not match " $expectedName ". Please Refer to csi-rclone documentation." }}
+{{- end }}
```

Comment:
It makes sense to run this check also only when we deploy the csi rclone chart as part of amalthea. If not then we should not check. So I just add an outer if statement for this. You can use `and` but I dont remember the right syntax and you will need some brackets when you combine that with `not eq` in the same if statement. So I think the suggestion below is good enough and potentially clearer than the combined if statement.

```suggestion
{{- if .Values.deploy.csi-rclone }}
{{- if not eq .Values.rcloneStorageClass $expectedName }}
{{- fail "ERROR: .Values.rcloneStorageClass does not match " $expectedName ". Please Refer to csi-rclone documentation." }}
{{- end }}
{{- end }}
```